modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,345 @@
+
+    const [ firstPromise ] = editor.userLookup.fetchUsers([ userId ]);
+    const [ secondPromise ] = editor.userLookup.fetchUsers([ userId ]);
+
+    await expect(firstPromise).to.eventually.deep.equal(
+      await secondPromise,
+      'Should return same data for subsequent requests'
+    );
+  });
+
+  it('TINY-11974: Should reject promise for non-existent users', async () => {
+    const editor = hook.editor();
+    const nonExistentId = 'non-existent-user';
+
+    const [ userPromise ] = editor.userLookup.fetchUsers([ nonExistentId ]);
+
+    await expect(userPromise).to.eventually.be.rejectedWith(
+      Error,
+      `User ${nonExistentId} not found`,
+      'Should reject for non-existent user'
+    );
+  });
+
+  it('TINY-11974: Should handle repeated requests for non-existent users', async () => {
+    const editor = hook.editor();
+    const nonExistentId = 'non-existent-user';
+    const errorMessage = `User ${nonExistentId} not found`;
+
+    const [ firstPromise ] = editor.userLookup.fetchUsers([ nonExistentId ]);
+    await expect(firstPromise).to.eventually.be.rejectedWith(errorMessage);
+
+    // Second request should use cached rejection
+    const [ secondPromise ] = editor.userLookup.fetchUsers([ nonExistentId ]);
+    await expect(secondPromise).to.eventually.be.rejectedWith(errorMessage);
+  });
+
+  it('TINY-11974: Should handle mix of existing and non-existent users', async () => {
+    const editor = hook.editor();
+    const existingId = 'test-user-1';
+    const nonExistentId = 'non-existent-user';
+
+    const [ existingPromise, nonExistentPromise ] = editor.userLookup.fetchUsers([ existingId, nonExistentId ]);
+
+    await expect(existingPromise).to.eventually.deep.equal({
+      id: existingId,
+      name: 'Test User',
+      avatar: 'test-avatar.png',
+      description: 'Test Description'
+    });
+
+    await expect(nonExistentPromise).to.be.rejectedWith(`User ${nonExistentId} not found`);
+  });
+
+  it('TINY-11974: Should handle concurrent requests for the same user', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    // Make multiple concurrent requests
+    const [ promise1, promise2, promise3 ] = [
+      editor.userLookup.fetchUsers([ userId ])[0],
+      editor.userLookup.fetchUsers([ userId ])[0],
+      editor.userLookup.fetchUsers([ userId ])[0]
+    ];
+
+    await Promise.all([
+      expect(promise1).to.eventually.deep.equal(await promise2),
+      expect(promise2).to.eventually.deep.equal(await promise3),
+    ]);
+  });
+
+  it('TINY-11974: Should handle invalid user IDs gracefully', async () => {
+    const editor = hook.editor();
+    const invalidIds = [ '', ' ', undefined, null, false, 123 ] as any[];
+
+    // Test each invalid ID asynchronously
+    await Promise.all(Arr.map(invalidIds, async (invalidId) => {
+      const [ promise ] = editor.userLookup.fetchUsers([ invalidId ]);
+      await expect(promise).to.be.rejectedWith(
```

Comment:
Maybe we need to run some promise tutorials?
```suggestion
      return expect(promise).to.be.rejectedWith(
```

Reply:
Nope, that's just me forgetting to remove things. The `Promise.all` is already handling the awaiting so the `await` on the inner `expect` is redundant. We would also need to remove the `async` keyword from the function provided to `map`.